schema/data.fbs

```diff
@@ -0,0 +1,83 @@
+include "scalar_type.fbs";
+namespace executorch_flatbuffer;
+
+// Update after BC breaking changes.
+file_identifier "DT01";
+file_extension "data";
+
+table TensorMetadata {
+  // The unique id used to connect the data and program.
+  fully_qualified_name:string;
+  scalar_type:ScalarType;
+
+  // Size of each dimension.
+  dim_sizes:[int];
```

Comment:
equivalent of 'sizes' in `program.fbs`, using `dim_sizes` to differentiate from `size` of the tensor below.